### PR TITLE
Hd Abhishek-21 Mar

### PR DESCRIPTION
**Pull request by gem-abhishek-gautam**
@Gem-NeerajSingh PR ready for review

### CODE REVIEW

File: src/test/java/com/qa/helpdesk/stepDefinitions/FilterAndPageStepDefinition.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+            }
+            if(!status.isBlank())
+            {
+                DriverAction.click(FilterAndCalendarLocators.filterOptions(status));
+                DriverAction.waitSec(3);
+
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```
Comment (Gem-NeerajSingh) [inline src/test/java/com/qa/helpdesk/stepDefinitions/FilterAndPageStepDefinition.java:75]: Add gemtest reporter

File: src/test/java/com/qa/helpdesk/stepDefinitions/FilterAndPageStepDefinition.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+                    if(statusFlag==1){
+                        GemTestReporter.addTestStep("Status filter","Status filter results are as expected",STATUS.PASS,DriverAction.takeSnapShot());
+                    } else GemTestReporter.addTestStep("Status filter","Status filter results NOT as expected",STATUS.FAIL,DriverAction.takeSnapShot());
+
+                } else GemTestReporter.addTestStep("Status filter","No results for filter status "+status+"",STATUS.INFO,DriverAction.takeSnapShot());
+
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
```
Comment (Gem-NeerajSingh) [inline src/test/java/com/qa/helpdesk/stepDefinitions/FilterAndPageStepDefinition.java:124]: Add gemtest reporter

File: src/test/java/com/qa/helpdesk/stepDefinitions/HeaderButtonStepDefinitions.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+            DriverAction.waitSec(5);
+            if(DriverAction.isExist(DashboardHeaderLocators.pdfHeader) && DriverAction.getElementText(DashboardHeaderLocators.pdfHeader).equalsIgnoreCase("Helpdesk user guide.pdf"))
+            {
+                GemTestReporter.addTestStep("Verify user-guide PDF","User-guide pdf is displayed", STATUS.PASS,DriverAction.takeSnapShot());
+            } else GemTestReporter.addTestStep("Verify user-guide PDF","User-guide pdf is not displayed", STATUS.FAIL,DriverAction.takeSnapShot());
+
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
```
Comment (Gem-NeerajSingh) [inline src/test/java/com/qa/helpdesk/stepDefinitions/HeaderButtonStepDefinitions.java:45]: Add gemtest reporter

File: src/test/java/com/qa/helpdesk/stepDefinitions/HeaderButtonStepDefinitions.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+    }
+
+    @Given("Click on support button")
+    public void clickOnSupportButton() {
+        try {
+            DriverAction.click(DashboardHeaderLocators.headerButtons("Support"));
+            DriverAction.waitSec(2);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```
Comment (Gem-NeerajSingh) [inline src/test/java/com/qa/helpdesk/stepDefinitions/HeaderButtonStepDefinitions.java:55]: Add gemtest reporter

File: src/test/java/com/qa/helpdesk/stepDefinitions/HeaderButtonStepDefinitions.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+    @Then("Verify if support info is displayed")
+    public void verifyIfSupportInfoIsDisplayed() {
+        try {
+            if(DriverAction.getElementText(DashboardHeaderLocators.contact).equalsIgnoreCase("support.helpdesk@geminisolutions.com")) {
+                GemTestReporter.addTestStep("Contact info","Email address visible",STATUS.PASS,DriverAction.takeSnapShot());
+            } else GemTestReporter.addTestStep("Contact info","Email address is not visible Expected: support.helpdesk@geminisolutions.com",STATUS.FAIL,DriverAction.takeSnapShot());
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
```
Comment (Gem-NeerajSingh) [inline src/test/java/com/qa/helpdesk/stepDefinitions/HeaderButtonStepDefinitions.java:67]: Add gemtest reporter

File: src/test/java/com/qa/helpdesk/stepDefinitions/HeaderButtonStepDefinitions.java [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+    public void clickOnNotificationButton() {
+        try {
+            DriverAction.waitUntilElementAppear(DashboardHeaderLocators.notificationButton,10);
+            if(DriverAction.isExist(DashboardHeaderLocators.notificationButton)) {
+                DriverAction.click(DashboardHeaderLocators.notificationButton,"Notification button");
+            } else GemTestReporter.addTestStep("Notification button","Notification button not visible", STATUS.FAIL, DriverAction.takeSnapShot());
+            DriverAction.waitSec(2);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
```
Comment (Gem-NeerajSingh) [inline src/test/java/com/qa/helpdesk/stepDefinitions/HeaderButtonStepDefinitions.java:80]: Add gemtest reporter

File: src/test/java/com/qa/helpdesk/stepDefinitions/HeaderButtonStepDefinitions.java [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+
+    @Then("Check if notifications are displayed")
+    public void checkIfNotificationsAreDisplayed() {
+        try {
+            if(DriverAction.isExist(DashboardHeaderLocators.notificationContainer)){
+                GemTestReporter.addTestStep("Notifications","Notification container is visible", STATUS.PASS, DriverAction.takeSnapShot());
+            } else GemTestReporter.addTestStep("Notifications","Notification container not visible", STATUS.FAIL, DriverAction.takeSnapShot());
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
```
Comment (Gem-NeerajSingh) [inline src/test/java/com/qa/helpdesk/stepDefinitions/HeaderButtonStepDefinitions.java:91]: Add gemtest reporter

File: src/test/java/com/qa/helpdesk/stepDefinitions/HeaderButtonStepDefinitions.java [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+                DriverAction.click(DashboardHeaderLocators.notificationButton,"Notification button");
+                DriverAction.waitSec(2);
+                DriverAction.click(DashboardHeaderLocators.unreadNotifications,"Unread");
+                DriverAction.waitSec(2);
+                DriverAction.click(DashboardHeaderLocators.notificationReadButton,"Mark as read");
+                DriverAction.waitSec(2);
+            } GemTestReporter.addTestStep("Notifications","No unread count", STATUS.INFO, DriverAction.takeSnapShot());
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
```
Comment (Gem-NeerajSingh) [inline src/test/java/com/qa/helpdesk/stepDefinitions/HeaderButtonStepDefinitions.java:107]: Add gemtest reporter

File: src/test/java/com/qa/helpdesk/stepDefinitions/HeaderButtonStepDefinitions.java [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+            DriverAction.waitSec(1);
+            DriverAction.click(DashboardHeaderLocators.unreadNotifications,"Unread");
+            DriverAction.waitSec(1);
+            if(!DriverAction.isExist(DashboardHeaderLocators.unreadCount) && DriverAction.getElementText(DashboardHeaderLocators.notificationListAlert).contains("No new")) {
+                GemTestReporter.addTestStep("Notification read","Notifications mark as read",STATUS.PASS,DriverAction.takeSnapShot());
+            } else GemTestReporter.addTestStep("Notification read","Notifications NOT mark as read",STATUS.FAIL,DriverAction.takeSnapShot());
+
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
```
Comment (Gem-NeerajSingh) [inline src/test/java/com/qa/helpdesk/stepDefinitions/HeaderButtonStepDefinitions.java:123]: Add gemtest reporter

File: src/test/java/com/qa/helpdesk/stepDefinitions/Hook.java [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+public class Hook {
+    @Before
+    public void setup() throws GemException {
+//        ChromeOptions options = new ChromeOptions();
+//        options.addArguments("--remote-allow-origins=*");
+//        options.addArguments("--incognito");
+//        DriverManager.initializeChrome(options);
+//        DriverAction.launchUrl("https://helpdeskui-np.geminisolutions.com/");
+//        DriverAction.maximizeBrowser();
+            DriverManager.setUpBrowser();
```
Comment (Gem-NeerajSingh) [inline src/test/java/com/qa/helpdesk/stepDefinitions/Hook.java:18]: remove dead code

File: src/test/java/com/qa/helpdesk/stepDefinitions/SearchAndSortStepDefinition.java [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+                }
+
+                if(itemFoundFlag==1){
+                    GemTestReporter.addTestStep("Ticket search result","Ticket search result is as expected", STATUS.PASS,DriverAction.takeSnapShot());
+                } else GemTestReporter.addTestStep("Ticket search result","Ticket search result is not as expected",STATUS.FAIL,DriverAction.takeSnapShot());
+
+            } else GemTestReporter.addTestStep("Ticket search result","No search results for the keyword "+keyword+"",STATUS.INFO,DriverAction.takeSnapShot());
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
```
Comment (Gem-NeerajSingh) [inline src/test/java/com/qa/helpdesk/stepDefinitions/SearchAndSortStepDefinition.java:56]: Add gemtest reporter

File: src/test/resources/features/HeaderButtons.feature [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+    Given Click on user guide button
+    Then Verify if user guide is opened
+
+  Scenario: Check support button functionality
+    Given Click on support button
+    Then Verify if support info is displayed
+
+  Scenario: Check notification button
+    Given Click on notification button
+    Then Check if notifications are displayed
```
Comment (Gem-NeerajSingh) [inline src/test/resources/features/HeaderButtons.feature:17]: replace check with verify

File: src/test/resources/features/CreateTicket.feature [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+  Scenario Outline: Verify Request ticket details
+    Given create request without file and with subject "<subject>", dept "<dept>", category "<catg>", sub-category "<sub>" and description "<desc>"
+    Then submit and verify the request details for subject "<subject>", dept "<dept>", description "<desc>", category "<catg>" and sub-category "<sub>"
+    Examples:
+      | subject     | dept | desc                |catg|sub|
+      |test request|Admin|This is a test ticket |Facility|Chair|
+
+  Scenario Outline: Check incident ticket cancellation functionality
+    Given create incident without file and subject "<subject>", dept "<dept>" and description "<desc>"
+    Then submit and cancel the ticket with reason "<reason>"
```
Comment (Gem-NeerajSingh) [inline src/test/resources/features/CreateTicket.feature:52]: Then submit and cancel the ticket with reason "<reason>"
change to and

File: src/test/java/com/qa/helpdesk/stepDefinitions/FilterAndPageStepDefinition.java [hunk 14]
```diff
@@ -0,0 +1,130 @@
...
+
+    @Then("Verify result for selection {string}")
+    public void verifyResultForSelection(String option) {
+        try {
+            List<WebElement> elements = DriverAction.getElements(TableAndPaginationLocators.getTableRows);
+            if(elements.size()==Integer.parseInt(option)){
+                GemTestReporter.addTestStep("Pagination","Rows per page displayed as expected",STATUS.PASS,DriverAction.takeSnapShot());
+            } else GemTestReporter.addTestStep("Pagination","Rows per page displayed NOT as expected",STATUS.FAIL,DriverAction.takeSnapShot());
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```
Comment (Gem-NeerajSingh) [inline src/test/java/com/qa/helpdesk/stepDefinitions/FilterAndPageStepDefinition.java:40]: Add gemtest reporter

File: src/test/resources/features/SearchAndSort.feature [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+    Given Search for keyword "<keyword>"
+    Then Verify search result for "<keyword>"
+    Examples:
+      | keyword |
+      |INC-000000591|
+      |request|
+      |test   |
+
+  Scenario: Verify sorting buttons functionality
+    Given Verify sorting button for "ID" column
```
Comment (Gem-NeerajSingh) [inline src/test/resources/features/SearchAndSort.feature:18]: Marking this scenario for my reference will update on this later

**Review by Gem-NeerajSingh — CHANGES_REQUESTED** [review]
Request changes

**Comment by gem-abhishek-gautam** [comment]
done

**Pull request closed without merging**